src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -124,14 +147,36 @@ const InfiniteScroll = ({
         Math.max(
           (!replace && endPage) || 0,
           multiColumn
-            ? Math.ceil(((top + height + offset) * width) / pageArea)
-            : Math.floor((top + height + offset) / pageHeight),
+            ? Math.ceil((scrollBufferBottom * width) / avgPageArea)
+            : Math.floor(scrollBufferBottom / avgPageHeight),
           show ? Math.floor(show / step) : 0,
         ),
       );
-
       if (nextBeginPage !== beginPage) setBeginPage(nextBeginPage);
       if (nextEndPage !== endPage) setEndPage(nextEndPage);
+
+      // Keep track of page heights as begin/end pages increment/decrement
```

Comment:
Would it be better to keep all page heights in this array instead of removing? Then we could reference the avgPageHeight calculation by beginPage and endPage indexes in which we are interested?

Reply:
just double checking the only piece of this code block you want to keep is 
```
    if (nextEndPage > endPage) {
        nextPageHeights.push(pageHeight);
        nextPageAreas.push(pageArea);
      }
```
?